server/sonar-server/src/main/java/org/sonar/server/user/ThreadLocalUserSession.java

```diff
@@ -31,26 +31,26 @@
  */
 public class ThreadLocalUserSession implements UserSession {
 
-  private static final ThreadLocal<UserSession> THREAD_LOCAL = new ThreadLocal<>();
+  private final ThreadLocal<UserSession> delegate = new ThreadLocal<>();
```

Comment:
ThreadLocal instances should be static. It's almost a none sense to have them non static.
http://stackoverflow.com/questions/2784009/why-should-java-threadlocal-variables-be-static